sdk/src/logger.rs

```diff
@@ -1,51 +1,84 @@
 
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
 pub struct Logger {
-    pub target: &'static str,
     pub id: String,
 }
 
+impl From<&str> for Logger {
+    fn from(id: &str) -> Self {
+        Logger::from(id.to_string())
+    }
+}
+
+impl From<String> for Logger {
+    fn from(id: String) -> Self {
+        Logger { id }
+    }
+}
+
 impl Logger {
+    /// Calls the [`log::trace!`] macro, using the logger's configurations.
     pub fn trace<T: Display>(&self, msg: T) {
         if self.id.is_empty() {
-            log::trace!(target: self.target, "{msg}");
+            trace!(target: "gadget", "{msg}");
             return;
         }
-        log::trace!(target: self.target, "[{}] {msg}", &self.id);
+        trace!(target: "gadget", "[{}] {msg}", &self.id);
     }
 
+    /// Calls the [`log::debug!`] macro, using the logger's configurations.
     pub fn debug<T: Display>(&self, msg: T) {
         if self.id.is_empty() {
-            log::debug!(target: self.target, "{msg}");
+            debug!(target: "gadget", "{msg}");
             return;
         }
-        log::debug!(target: self.target, "[{}] {msg}", &self.id);
+        debug!(target: "gadget", "[{}] {msg}", &self.id);
     }
 
+    /// Calls the [`log::info!`] macro, using the logger's configurations.
     pub fn info<T: Display>(&self, msg: T) {
         if self.id.is_empty() {
-            log::info!(target: self.target, "{msg}");
+            info!(target: "gadget", "{msg}");
             return;
         }
-        log::info!(target: self.target, "[{}] {msg}", &self.id);
+        info!(target: "gadget", "[{}] {msg}", &self.id);
     }
 
+    /// Calls the [`log::warn!`] macro, using the logger's configurations.
     pub fn warn<T: Display>(&self, msg: T) {
         if self.id.is_empty() {
-            log::warn!(target: self.target, "{msg}");
+            warn!(target: "gadget", "{msg}");
             return;
         }
-        log::warn!(target: self.target, "[{}] {msg}", &self.id);
+        warn!(target: "gadget", "[{}] {msg}", &self.id);
     }
 
+    /// Calls the [`log::error!`] macro, using the logger's configurations.
     pub fn error<T: Display>(&self, msg: T) {
         if self.id.is_empty() {
-            log::error!(target: self.target, "{msg}");
+            error!(target: "gadget", "{msg}");
             return;
         }
-        log::error!(target: self.target, "[{}] {msg}", &self.id);
+        error!(target: "gadget", "[{}] {msg}", &self.id);
     }
 }
+
+pub fn setup_log() {
+    use tracing_subscriber::fmt::SubscriberBuilder;
+    use tracing_subscriber::util::SubscriberInitExt;
+    use tracing_subscriber::EnvFilter;
+
+    let _ = SubscriberBuilder::default()
+        .with_env_filter(EnvFilter::from_default_env())
+        .finish()
+        .try_init();
+
+    std::panic::set_hook(Box::new(|info| {
+        log::error!(target: "gadget", "Panic occurred: {info:?}");
+    }));
+}
```

Comment:
```suggestion
```

This should be removed.

* `tracing_subscriber` should only ever be used by applications
* There isn't any benefit to setting a new panic hook here. It's actually a *loss* of information, since you no longer get the backtrace.